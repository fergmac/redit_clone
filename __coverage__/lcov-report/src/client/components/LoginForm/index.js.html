<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/components/LoginForm/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/client/components/LoginForm</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/13</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent } from 'r<span class="cstat-no" title="statement not covered" >eact';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
import { Field, reduxForm } from 'redux-form';
import styles from './style.css';
import { connect } from 'react-redux';
import { saveUser } from '../../../shared/redux/modules/users';
import { TextField } from 'redux-form-material-ui';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
const validate <span class="cstat-no" title="statement not covered" >= values <span class="fstat-no" title="function not covered" >=&gt; {</span></span>
  const errors <span class="cstat-no" title="statement not covered" >= {</span>};
  const requiredFields <span class="cstat-no" title="statement not covered" >= ['email', 'password'</span>];
<span class="cstat-no" title="statement not covered" >  requiredFields.forEach(<span class="fstat-no" title="function not covered" >fi</span>eld =&gt; {</span>
<span class="cstat-no" title="statement not covered" >    if (!values[field]) {</span>
<span class="cstat-no" title="statement not covered" >      errors[field] = 'Required';</span>
    }
  });
<span class="cstat-no" title="statement not covered" >  if (values.email &amp;&amp; !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {</span>
<span class="cstat-no" title="statement not covered" >    errors.email = 'Invalid email address';</span>
  }
<span class="cstat-no" title="statement not covered" >  return errors;</span>
};
&nbsp;
class LoginForm extends PureComponent {
  componentDidMount() {
    this.props.saveUser({
      email: 'f.macconnell@gmail.com',
      password: 'password',
    });
    // this.refs.name
    //   .getRenderedComponent()
    //   .getRenderedComponent()
    //   .focus();
  }
  render() {
    const { handleSubmit, pristine, reset, submitting } = this.props;
    return (
      &lt;div className={styles.loginFormPage}&gt;
        &lt;h1&gt;LoginForm&lt;/h1&gt;
        &lt;form onSubmit={handleSubmit}&gt;
          &lt;Field name="email" component={TextField} type="text" placeholder="email" /&gt;
          &lt;Field name="password" component={TextField} type="password" placeholder="password" /&gt;
          &lt;div className={styles.loginButtons}&gt;
            &lt;button type="submit" disabled={pristine || submitting}&gt;Login&lt;/button&gt;
            &lt;button type="submit" disabled={pristine || submitting} onClick={reset}&gt;Sign Up&lt;/button&gt;
          &lt;/div&gt;
        &lt;/form&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps <span class="cstat-no" title="statement not covered" >= state =<span class="fstat-no" title="function not covered" >&gt; ({</span></span>
  users: state.users,
});
&nbsp;
const mapDispatchToProps <span class="cstat-no" title="statement not covered" >= (dispat<span class="fstat-no" title="function not covered" >ch) =&gt; ({</span></span>
  saveUser: (newUser) =&gt; {
    dispatch(saveUser(newUser));
  },
});
&nbsp;
&nbsp;
// export default reduxForm({
//   form: 'login',
//   validate,
// })(LoginForm);
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'login',
  validate,
})(LoginForm));
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Nov 09 2016 16:25:23 GMT-0800 (PST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
