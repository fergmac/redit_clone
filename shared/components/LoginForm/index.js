import React from 'react';
import styles from '../../../web.browser/styles/shared.css';

const LoginForm = () => (
  <div className={styles.loginFormPage}>
    <h1>Login Form</h1>
  </div>
);

export default LoginForm;
