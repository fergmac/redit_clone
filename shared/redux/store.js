import { createStore } from 'redux';
import reducers from './reducers';

const store = createStore(reducers, {});

console.log(store);

export default store;
