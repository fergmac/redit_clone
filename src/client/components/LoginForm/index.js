import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form';
import styles from './style.css';
import { connect } from 'react-redux';
import { saveUser } from '../../../shared/redux/modules/users';
import { TextField } from 'redux-form-material-ui';

const validate = values => {
  const errors = {};
  const requiredFields = ['email', 'password'];
  requiredFields.forEach(field => {
    if (!values[field]) {
      errors[field] = 'Required';
    }
  });
  if (values.email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  return errors;
};

class LoginForm extends PureComponent {
  componentDidMount() {
    this.props.saveUser({
      email: 'f.macconnell@gmail.com',
      password: 'password',
    });
    // this.refs.name
    //   .getRenderedComponent()
    //   .getRenderedComponent()
    //   .focus();
  }
  render() {
    const { handleSubmit, pristine, reset, submitting } = this.props;
    return (
      <div className={styles.loginFormPage}>
        <h1>LoginForm</h1>
        <form onSubmit={handleSubmit}>
          <Field name="email" component={TextField} type="text" placeholder="email" />
          <Field name="password" component={TextField} type="password" placeholder="password" />
          <div className={styles.loginButtons}>
            <button type="submit" disabled={pristine || submitting}>Login</button>
            <button type="submit" disabled={pristine || submitting} onClick={reset}>Sign Up</button>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  users: state.users,
});

const mapDispatchToProps = (dispatch) => ({
  saveUser: (newUser) => {
    dispatch(saveUser(newUser));
  },
});


// export default reduxForm({
//   form: 'login',
//   validate,
// })(LoginForm);

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'login',
  validate,
})(LoginForm));
