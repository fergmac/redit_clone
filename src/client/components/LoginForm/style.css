.loginFormPage {
  display: flex;
  width: 100%;
  height: 100%;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
}

.loginFormPage form{
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
}
 input {
  padding: 5px 50px;
  margin: 5px 0px;
}

.loginButtons {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  margin: 5px 0px;
}