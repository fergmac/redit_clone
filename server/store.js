import { createStore } from 'redux';
import reducers from '../src/shared/redux/reducers';

export default createStore(reducers, {});


