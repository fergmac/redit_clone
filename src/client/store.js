import { createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import reducers from '../shared/redux/reducers';
import remoteMiddleware from './socket/remoteMiddleware';
import socket from './socket';

const logger = createLogger();

const middlewares = applyMiddleware(reduxThunk, remoteMiddleware(socket), logger);

const store = createStore(reducers, middlewares);

export default store;
