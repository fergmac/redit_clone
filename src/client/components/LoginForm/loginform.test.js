/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoginForm from './index';
import { Provider } from 'react-redux';
import store from '../../store';


describe('LoginForm', () => {
    const userData = {
        
    }
    it('creates a new user', () => {
       const wrapper = shallow(
           <Provider store={store}>
                <LoginForm 
                    // album={albumData}
                />
          </Provider>
       )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})