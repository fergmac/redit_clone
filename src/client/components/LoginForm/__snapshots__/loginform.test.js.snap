exports[`LoginForm creates a new user 1`] = `<Connect(ReduxForm) />`;
